### PR TITLE
nixos/murmur: expose dbus

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Murmur has three RPC interfaces currently:
- gRPC, which is getting removed in the next release
- ZeroC ICE, which is a gigantic pain to use
- D-Bus, which is considered deprecated, but tolerable

This at least allows enabling D-Bus.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/murmur.nix [hunk 1]
```diff
@@ -282,6 +284,12 @@ in
           `murmur` is running.
         '';
       };
+
+      dbus = mkOption {
+        type = types.enum [ null "session" "system" ];
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/murmur.nix:289]: Shouldn't this be nullOr types.enum?

File: nixos/modules/services/networking/murmur.nix [hunk 2]
```diff
@@ -325,5 +333,25 @@ in
...
+          </policy>
+
+          <policy context="default">
+            <allow send_destination="net.sourceforge.mumble.murmur"/>
+            <allow receive_sender="net.sourceforge.mumble.murmur"/>
+          </policy>
+        </busconfig>
+      '';
+      destination = "/share/dbus-1/system.d/murmur.conf";
+    })];
```
[the remark this answers is not in the document]
Comment (K900) [reply nixos/modules/services/networking/murmur.nix:355]: We could just put it in $out directly really, but I'm not sure how I feel about that given upstream doesn't ship this...

[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply nixos/modules/services/networking/murmur.nix:355]: > We could just put it in $out directly really, but I'm not sure how I feel about that given upstream doesn't ship this...

Did you raise it to them? (that they do not ship it?)

[the remark this answers is not in the document]
Comment (K900) [reply nixos/modules/services/networking/murmur.nix:355]: No, but they also consider DBus deprecated so I doubt they'll do anything about it. Might be worth trying though.

**Pull request merged**